templates/federated_reporting/distributed_cleanup.py

```diff
@@ -226,7 +228,7 @@ def main():
     logger.addHandler(ch)
 
     if not os.path.exists(DISTRIBUTED_CLEANUP_SECRET_PATH):
-        if sys.stdout.isatty():
+        if sys.stdout.isatty() or args.stdin:
```

Comment:
`getpass()` doesn't read stuff from stdin on my system:
```
echo blah | python3 -c 'from getpass import getpass; p = getpass(); print(p)'
Password: 
```

Reply:
mmm, good catch, I was pretty sure it worked for me, will check again before more CI.

Reply:
In the tests so far I think getpass is operating "properly" aka not hanging and just waiting and issuing the warning the docs mention

```If echo free input is unavailable getpass() falls back to printing a warning message to stream and reading from sys.stdin and issuing a [GetPassWarning](https://docs.python.org/3/library/getpass.html#getpass.GetPassWarning).```

https://docs.python.org/3/library/getpass.html

Pretty sure this aspect of things are working in the test: https://ci.cfengine.com/job/fr-tests/4558/testReport/junit/(root)/new_tests_FR_tests_all_platforms_/FR_distributed_cleanup_test/

```+ ssh -o LogLevel=ERROR -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o ConnectTimeout=60 root@54.228.138.210 python3 /opt/cfengine/federation/bin/distributed_cleanup.py --stdin
/usr/lib64/python3.6/getpass.py:91: GetPassWarning: Can not control echo on the terminal.
  passwd = fallback_getpass(prompt, stream)
Warning: Password input may be echoed.
Enter admin password for superhub ip-172-31-33-213.eu-west-1.compute.internal: Enter email for fr_distributed_cleanup accounts: 
/usr/lib64/python3.6/getpass.py:91: GetPassWarning: Can not control echo on the terminal.
  passwd = fallback_getpass(prompt, stream)
Warning: Password input may be echoed.
Enter admin credentials for feeder0: 
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/urllib3/util/ssl_.py", line 402, in ssl_wrap_socket
    context.load_verify_locations(ca_certs, ca_cert_dir, ca_cert_data)
ssl.SSLError: unknown error (_ssl.c:3762)

During handling of the above exception, another exception occurred:
```

The ssl error is unrelated to the getpass issue.

Reply:
Cool. I guess it behaves differently because it doesn't have access to a PTY the way we run it. :+1:

Reply:
Yes, I think that must be the case.